### PR TITLE
:bug: (RUMM-285): Prevent concurrent access to SimpleDateFormat

### PR DESCRIPTION
**Pull request by xgouchet**
### What does this PR do?

Prevent crashing when using SimpleDateFormat

**This is not tested as there's no way to consistently reproduce this error in a test**

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [x] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [x] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/net/NetworkTimeInterceptor.kt [hunk 1]
```diff
@@ -43,7 +43,9 @@ internal class NetworkTimeInterceptor(
 
         val serverDateStr = response.header("date").orEmpty()
         val serverDate = try {
-            formatter.parse(serverDateStr)
+            synchronized(this) {
```
Comment (mariusc83) [inline dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/net/NetworkTimeInterceptor.kt]: That's interesting

Comment (mariusc83) [reply dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/net/NetworkTimeInterceptor.kt]: I am thinking maybe we should synchronize on the formatter instead ...

File: dd-sdk-android/src/main/kotlin/com/datadog/android/log/internal/domain/LogSerializer.kt [hunk 2]
```diff
@@ -51,7 +51,9 @@ internal class LogSerializer(private val logConstraints: LogConstraints = Datado
         jsonLog.addProperty(TAG_APP_PACKAGE_NAME, CoreFeature.packageName)
 
         // Timestamp
-        val formattedDate = simpleDateFormat.format(Date(log.timestamp))
+        val formattedDate = synchronized(this) {
+            simpleDateFormat.format(Date(log.timestamp))
```
Comment (mariusc83) [inline dd-sdk-android/src/main/kotlin/com/datadog/android/log/internal/domain/LogSerializer.kt]: Same here...

File: dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/net/NetworkTimeInterceptor.kt (hunk 1 above)
Comment (xgouchet) [reply dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/net/NetworkTimeInterceptor.kt]: That's a thought

**Pull request merged**